core/matrix/coo.cpp

```diff
@@ -47,6 +48,23 @@ GKO_REGISTER_OPERATION(aos_to_soa, components::aos_to_soa);
 }  // anonymous namespace
 }  // namespace coo
 
+template <typename ValueType, typename IndexType>
+void Coo<ValueType, IndexType>::validate_data() const
+{
+    GKO_VALIDATE(validation::is_sorted(row_idxs_),
+                 "row_idxs must be non-decending");
+
+    GKO_VALIDATE(
+        validation::is_within_bounds(row_idxs_, 0, this->get_size()[0]),
+        "row_idxs must be within bounds");
+
+    GKO_VALIDATE(
+        validation::is_within_bounds(col_idxs_, 0, this->get_size()[1]),
+        "col_idxs must be within bounds");
+
+    GKO_VALIDATE(validation::is_finite(values_),
+                 "matrix must contain only finite values");
```

Comment:
I don't think we need to separate them by empty lines, also we can simplify the code a bit
```suggestion
    using namespace gko::validation;
    GKO_VALIDATE(is_sorted(row_idxs_), "row_idxs must be non-decending");
    GKO_VALIDATE(is_within_bounds(row_idxs_, 0, this->get_size()[0]), "row_idxs must be within bounds");
    GKO_VALIDATE(is_within_bounds(col_idxs_, 0, this->get_size()[1]), "col_idxs must be within bounds");
    GKO_VALIDATE(is_finite(values_), "matrix must contain only finite values");
```